packages/easy-testing/src/DataProvider/StaticFixtureFinder.php

```diff
@@ -13,24 +13,18 @@ final class StaticFixtureFinder
 {
     public static function yieldDirectory(string $directory, string $suffix = '*.php.inc'): Iterator
     {
-        $fileInfos = self::findFilesInDirectory($directory, $suffix);
+        $finder = self::findFilesInDirectory($directory, $suffix);
 
-        foreach ($fileInfos as $fileInfo) {
-            yield [new SmartFileInfo($fileInfo->getRealPath())];
+        foreach ($finder as $fileInfo) {
+            yield $fileInfo->getRelativePathname() => [new SmartFileInfo($fileInfo->getRealPath())];
         }
     }
 
-    /**
-     * @return SplFileInfo[]
-     */
-    private static function findFilesInDirectory(string $directory, string $suffix): array
+    private static function findFilesInDirectory(string $directory, string $suffix): Finder
```

Comment:
I'm still thinking about this. If it would be useful to PHP community, it would be worth it to have it here.

This service can be extended with a new method e.g. `yieldDirectoryByFileName()` with proposed behavior. That way you could choose the method you prefer. What do you think?

Reply:
Hmm, that might be cool 👍 

A PR to phpunit would prob. be nice, so it supports both numbers and file names also, because then there is no need to differentiate.

Reply:
@TomasVotruba What do you like the most?
A new method or a new parameter?

Reply:
> A new method or a new parameter?

Better new method, because [optinal parameters hide the API](https://matthiasnoback.nl/2014/04/theres-no-such-thing-as-an-optional-dependency/)

> A PR to phpunit would prob. be nice, so it supports both numbers and file names also, because then there is no need to differentiate.

That would be amazing :+1:

Reply:
I have added a new method in 🥇 

Will add the pr to phpunit to my todo, let's see when I find time for that 🤣